### ISSUE TITLE
Lots of missing information

### ISSUE THREAD

**Issue by goyney** [veista/smartthings#20]
Thanks for the fork to try and get better support for the data exposed by Samsung. Once I installed this fork, I gained the ability to reset the Water Filter, but lost views into power consumption (over the stock integration). However, there is a TON more data that is exposed via SmartThings that just isn't showing up in Home Assistant with either the stock version or this fork.

![image](https://user-images.githubusercontent.com/1015040/169712370-d119274c-f8ca-4945-92dd-0567b327dfc9.png)

Stuff like being able to see/toggle the Rapid Cool/Freeze Modes would be nice. Also, the actual usage information from the Water Filter.

There are also child devices: 
![image](https://user-images.githubusercontent.com/1015040/169712416-72f2b21c-f686-407d-a370-75e5bef4eda5.png)

Having information about the Freezer (such as temps, etc.) would be helpful. Also, a toggle for the Ice Maker.

My specific fridge has an extra FlexZone drawer that has it's own contact sensor and the ability to change the "mode".
![image](https://user-images.githubusercontent.com/1015040/169712468-435a444e-d129-4842-a029-4afb830d2467.png)

These things should be controllable via the API since they're available in the SmartThings app:
![Screenshot_20220522-142622](https://user-images.githubusercontent.com/1015040/169712567-cfb76ab2-8bb9-40f0-b39e-353819ad8c0c.png)

Any way to help get all this extra stuff added?

**Comment by veista** [comment]
Hi,  
Thank you for your interest in the integration. I will add your model to the existing fridge entities. At this time, I am not adding new features. There will be an update later this year.

As for missing power consumption, that is weird.

**Comment by veista** [comment]
https://github.com/veista/smartthings/releases/tag/1.1.13

**Comment by goyney** [comment]
Thank you. That added a lot of the stuff I was looking for. The power information is still absent though.

**Comment by veista** [comment]
Please try removing the integration and adding it back again.

**Comment by goyney** [comment]
Nope, still no power information. It's alright, the stock version's didn't seem like it was updating anyways, so I don't know how much use it would have been to me anyways. I was just hoping to track consumption for my energy dashboard. Thanks!

**Comment by foltymat** [comment]
Hey. As I also have the same model fridge, I wanted to say thank you! Just a couple of things.

1. The energy monitoring would be super cool to have, if you have any free time to work on this :)
2. One bug I found is that the individual doors update to 'Open' but not back to 'Closed.' The Contact sensor works like a charm but not the individual doors.
3. For anyone looking into this thread, this particular fridge is in a 'Beta update' so if you're using HACS, you need to tourn on Beta versions.

Other thank that, it works perfectly fine. The setpoints can be adjusted as expected.

ref. 2)
![image](https://user-images.githubusercontent.com/62395302/173049293-7c09bf49-885c-44c6-bb09-21b7787f26ff.png)
